source/brailleDisplayDrivers/seika.py

```diff
@@ -57,12 +56,13 @@ def __init__(self):
 			# Read out the input buffer
 			versionS = self._ser.read(13)
 			log.debug("receive {p}".format(p=versionS))
-			if versionS.startswith("seika80"):
+			if versionS.startswith(b"seika80"):
 				log.info("Found Seika80 connected via {port} Version {versionS}".format(port=port, versionS=versionS))
 				self.numCells = 80
+				self.s40 = b"\xff\xff\x73\x38\x30\x00\x00\x00"

```

Comment:
Can you add a comment to explain what this is for, or what these values mean? This will be very helpful if the community has to maintain this file in the future.

Reply:
There needs handshake between Seika v3/5/80 and NVDA before any actual Braille data. The handshake data (\xff\xff\x73\x38\x30\x00\x00\x00) is one part of the transmit protocol of the Seika v3/5/80.

Reply:
Can you add this as a comment to the source code please.

Reply:
`s40` as a property name doesn't make much sense when it applies both to seika80 and 40.
Please rename to `sendHeader` based on the comment on line 105:
> `# every transmitted line consists of the preamble SEIKA_SENDHEADER and the Cells`

Looking at this again, I have realized what this encodes:
- Two bytes of `b'\xFF'`
- The code for the display `b"s80"`
- Null characters until the length is 8.

Note, this value:
``` python
>>> b"\xff\xff\x73\x38\x30\x00\x00\x00"
b'\xff\xffs80\x00\x00\x00'
>>> "s80".encode() == b"\x73\x38\x30"
True
```
Then from the later value where s40 is defined for 40 cell v3, v5.
``` python
>>> b"\xFF\xFF\x73\x65\x69\x6B\x61\x00"
b'\xff\xffseika\x00'
>>> b'\x73\x65\x69\x6B\x61' == b'seika'
True
```

It would be way clearer if these could be constructed: 
```python
def createDeviceVersionHandshake(deviceVersion:bytes)->bytes:
	""" Creates an 8 byte string containing the device version appropriately formatted for the handshake.
	if 5 < len(deviceVersion) < 1:
		raise ArgumentError("At least one character must be supplied. Only space in handshake for a total of 8 bytes. Requires 2 Leading 0xFF chars, and one trailing 0x0 char")
	totalChars = 8
	prefixChars = b"\xFF" * 2
	trailingChars = b"\x00" * (totalChars - len(prefixChars) - len(deviceVersion))
	return =  prefixChars + deviceVersion + trailingChars 
```

Then here you can do something like: 
```suggestion
				self.sendHeader= createDeviceVersionHandshake(b's80')

```

And a bit further down for the v3, v5:
``` python
				self.sendHeader=createDeviceVersionHandshake(b'seika')
```

Reply:
We have understood how to construct these headers, and what they encode. I would like this to be more obvious in the code.

Please add the following function:
```python
def createDeviceVersionHandshake(deviceVersion:bytes)->bytes:
	""" Creates an 8 byte string containing the device version appropriately formatted for the handshake.
	if 5 < len(deviceVersion) < 1:
		raise ArgumentError(
			"At least one character must be supplied."
			" Only space in handshake for a total of 8 bytes."
			" Requires 2 Leading 0xFF chars, and one trailing 0x0 char"
		)
	totalChars = 8
	prefixChars = b"\xFF" * 2
	data = prefixChars + deviceVersion
	# pad to required length
	data = data.ljust(totalChars, b'\x00')
	return data
```

Then where `b"\xff\xff\x73\x38\x30\x00\x00\x00"` is used change to `s80Header = createDeviceVersionHandshake(b"s80")`
and where `b"\xFF\xFF\x73\x65\x69\x6B\x61\x00"` is used change to `seikaHeader = createDeviceVersionHandshake(b"seika")`